diff --git a/piton/design/chipset/rtl/chipset.v b/piton/design/chipset/rtl/chipset.v
index 8ec4af5..1bed98f 100644
--- a/piton/design/chipset/rtl/chipset.v
+++ b/piton/design/chipset/rtl/chipset.v
@@ -85,6 +85,8 @@
 
 
 module chipset(
+  output corr_clk,
+  output mc_sys_clk_200MHz,
 
 `ifdef F1_BOARD
     input sys_clk,
@@ -226,7 +228,7 @@ module chipset(
 `ifdef PITONSYS_DDR4
     output                                      ddr_act_n,
     output [`DDR3_BG_WIDTH-1:0]                 ddr_bg,
-`else // PITONSYS_DDR4           
+`else // PITONSYS_DDR4
     output                                      ddr_cas_n,
     output                                      ddr_ras_n,
     output                                      ddr_we_n,
@@ -355,7 +357,7 @@ module chipset(
         inout                                           net_phy_mdio_io,
         output                                          net_phy_mdc,
 
-    `endif // PITON_FPGA_ETHERNETLITE    
+    `endif // PITON_FPGA_ETHERNETLITE
 `else // ifndef PITONSYS_IOCTRL
 
 `endif // endif PITONSYS_IOCTRL
@@ -455,7 +457,7 @@ module chipset(
         input                                               btnr,
         input                                               btnu,
         input                                               btnd,
-        input                                               btnc,           
+        input                                               btnc,
     `endif
 
     // Switches
@@ -464,13 +466,13 @@ module chipset(
         input  [3:0]                                        sw,
     `elsif XUPP3R_BOARD
         // no switches :(
-    `else         
+    `else
         input  [7:0]                                        sw,
     `endif
 
     `ifdef XUPP3R_BOARD
      output [3:0]                                           leds
-    `else 
+    `else
      output [7:0]                                           leds
      `endif
 
@@ -751,10 +753,10 @@ end
             `elsif XUPP3R_BOARD
                 assign uart_boot_en    = 1'b1;
                 assign uart_timeout_en = 1'b0;
-            `else 
+            `else
                 assign uart_boot_en    = sw[7];
                 assign uart_timeout_en = sw[6];
-            `endif    
+            `endif
         `endif // endif PITONSYS_UART_BOOT
     `endif // endif PITONSYS_UART
 `endif // endif PITONSYS_IOCTRL
@@ -766,9 +768,9 @@ end
     `elsif XUPP3R_BOARD
         // no switches :(
         assign noc_power_test_hop_count = 4'b0;
-    `else 
+    `else
         assign noc_power_test_hop_count = sw[3:0];
-    `endif    
+    `endif
 `endif // endif PITON_NOC_POWER_CHIPSET_TEST
 
 `ifdef PITON_BOARD
@@ -833,6 +835,7 @@ end
 //////////////////////////
 // Sub-module Instances //
 //////////////////////////
+assign mc_sys_clk_200MHz = mc_clk;
 
 // Clock generation
 `ifdef PITON_BOARD
@@ -859,6 +862,7 @@ end
         `ifdef PITON_CHIPSET_CLKS_GEN
             clk_mmcm    clk_mmcm    (
 
+
             `ifdef PITON_CHIPSET_DIFF_CLK
                 .clk_in1_p(clk_osc_p),
                 .clk_in1_n(clk_osc_n),
@@ -869,6 +873,8 @@ end
             .reset(1'b0),
             .locked(clk_locked),
 
+            .corr_clk(corr_clk),
+
             // Main chipset clock
             .chipset_clk(chipset_clk)
 
@@ -1244,9 +1250,9 @@ chipset_impl_noc_power_test  chipset_impl (
         `ifdef PITONSYS_DDR4
             .mc_clk_p(mc_clk_p),
             .mc_clk_n(mc_clk_n),
-        `else  // PITONSYS_DDR4                               
+        `else  // PITONSYS_DDR4
             .mc_clk(mc_clk),
-        `endif  // PITONSYS_DDR4                               
+        `endif  // PITONSYS_DDR4
     `endif // ifndef F1_BOARD
     `endif // endif PITON_FPGA_MC_DDR3
     `endif // endif PITONSYS_NO_MC
@@ -1273,13 +1279,13 @@ chipset_impl_noc_power_test  chipset_impl (
 
     // DRAM and I/O interfaces
     `ifndef PITONSYS_NO_MC
-        `ifdef PITON_FPGA_MC_DDR3 
+        `ifdef PITON_FPGA_MC_DDR3
             ,
             .init_calib_complete(init_calib_complete),
             `ifndef F1_BOARD
                 `ifdef PITONSYS_DDR4
-                    .ddr_act_n(ddr_act_n),                    
-                    .ddr_bg(ddr_bg), 
+                    .ddr_act_n(ddr_act_n),
+                    .ddr_bg(ddr_bg),
                 `else // PITONSYS_DDR4
                     .ddr_cas_n(ddr_cas_n),
                     .ddr_ras_n(ddr_ras_n),
@@ -1299,7 +1305,7 @@ chipset_impl_noc_power_test  chipset_impl (
                 `ifndef NEXYSVIDEO_BOARD
                     .ddr_cs_n(ddr_cs_n),
                 `endif // endif NEXYSVIDEO_BOARD
-            
+
                 `ifdef XUPP3R_BOARD
                     .ddr_parity(ddr_parity),
                 `else
@@ -1361,7 +1367,7 @@ chipset_impl_noc_power_test  chipset_impl (
                 .m_axi_bresp(m_axi_bresp),
                 .m_axi_buser(m_axi_buser),
                 .m_axi_bvalid(m_axi_bvalid),
-                .m_axi_bready(m_axi_bready), 
+                .m_axi_bready(m_axi_bready),
 
                 .ddr_ready(ddr_ready)
             `endif //ifndef F1_BOARD
@@ -1394,7 +1400,7 @@ chipset_impl_noc_power_test  chipset_impl (
             .sd_cmd(sd_cmd),
             .sd_dat(sd_dat)
         `endif // endif PITONSYS_SPI
-            `ifdef PITON_FPGA_ETHERNETLITE      
+            `ifdef PITON_FPGA_ETHERNETLITE
                 ,
                 .net_axi_clk        (net_axi_clk            ),
                 .net_phy_rst_n      (net_phy_rst_n          ),
@@ -1411,7 +1417,7 @@ chipset_impl_noc_power_test  chipset_impl (
                 .net_phy_mdio_io    (net_phy_mdio_io        ),
                 .net_phy_mdc        (net_phy_mdc            )
 
-            `endif // PITON_FPGA_ETHERNETLITE   
+            `endif // PITON_FPGA_ETHERNETLITE
     `endif // endif PITONSYS_IOCTRL
 
     `ifdef PITON_ARIANE
@@ -1561,7 +1567,7 @@ chipset_impl_noc_power_test  chipset_impl (
             net_phy_rxd_ff <= net_phy_rxd_f;
         end
 
-        assign net_phy_rxd_inter = net_phy_rxd_ff;                
+        assign net_phy_rxd_inter = net_phy_rxd_ff;
     `endif //PITON_FPGA_ETHERNETLITE
     //-------------------------------------------------------
 
@@ -1574,7 +1580,7 @@ chipset_impl_noc_power_test  chipset_impl (
             .D2(0),
             .SR(0)
             );
-    `else 
+    `else
         ODDR sd_clk_oddr (
             .Q(sd_clk_out),
             .C(sd_clk_out_internal),
