diff --git a/piton/design/xilinx/vc707/constraints.xdc b/piton/design/xilinx/vc707/constraints.xdc
index db5d307..2ab678c 100644
--- a/piton/design/xilinx/vc707/constraints.xdc
+++ b/piton/design/xilinx/vc707/constraints.xdc
@@ -11,7 +11,7 @@
 #     * Neither the name of Princeton University nor the
 #       names of its contributors may be used to endorse or promote products
 #       derived from this software without specific prior written permission.
-# 
+#
 # THIS SOFTWARE IS PROVIDED BY PRINCETON UNIVERSITY "AS IS" AND
 # ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
 # WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
@@ -32,8 +32,12 @@ set_property IOSTANDARD LVDS [get_ports chipset_clk_osc_n]
 set_property CLOCK_DEDICATED_ROUTE BACKBONE [get_nets chipset/clk_mmcm/inst/clk_in1_clk_mmcm]
 
 # Reset
-set_property IOSTANDARD LVCMOS18 [get_ports sys_rst_n]
-set_property PACKAGE_PIN AV40 [get_ports sys_rst_n]
+set_property IOSTANDARD LVCMOS18 [get_ports CPU_RESET]
+set_property PACKAGE_PIN AV40 [get_ports CPU_RESET]
+set_property IOSTANDARD LVCMOS18 [get_ports GPIO_SW_C]
+set_property PACKAGE_PIN AV39 [get_ports GPIO_SW_C]
+set_property IOSTANDARD LVCMOS18 [get_ports GPIO_SW_W]
+set_property PACKAGE_PIN AW40 [get_ports GPIO_SW_W]
 
 # False paths
 set_false_path -to [get_cells -hierarchical *afifo_ui_rst_r*]
@@ -154,6 +158,36 @@ set_input_delay -clock [get_clocks sd_clk_out_1] -clock_fall -max -add_delay 14.
 set_input_delay -clock [get_clocks sd_clk_out_1] -clock_fall -min -add_delay -14.000 [get_ports {sd_dat[*]}]
 set_input_delay -clock [get_clocks sd_clk_out_1] -clock_fall -min -add_delay -14.000 [get_ports sd_cmd]
 
+# {{{ FMC1
+# pwm
+set_property IOSTANDARD LVCMOS18 [get_ports FMC1_HPC_LA20_N]
+set_property PACKAGE_PIN Y30 [get_ports FMC1_HPC_LA20_N]
+set_property IOSTANDARD LVCMOS18 [get_ports FMC1_HPC_LA20_P]
+set_property PACKAGE_PIN Y29 [get_ports FMC1_HPC_LA20_P]
+set_property IOSTANDARD LVCMOS18 [get_ports FMC1_HPC_LA21_N]
+set_property PACKAGE_PIN N29 [get_ports FMC1_HPC_LA21_N]
+set_property IOSTANDARD LVCMOS18 [get_ports FMC1_HPC_LA21_P]
+set_property PACKAGE_PIN N28 [get_ports FMC1_HPC_LA21_P]
+set_property IOSTANDARD LVCMOS18 [get_ports FMC1_HPC_LA22_N]
+set_property PACKAGE_PIN P28 [get_ports FMC1_HPC_LA22_N]
+set_property IOSTANDARD LVCMOS18 [get_ports FMC1_HPC_LA22_P]
+set_property PACKAGE_PIN R28 [get_ports FMC1_HPC_LA22_P]
+set_property IOSTANDARD LVCMOS18 [get_ports FMC1_HPC_LA23_N]
+set_property PACKAGE_PIN N31 [get_ports FMC1_HPC_LA23_N]
+set_property IOSTANDARD LVCMOS18 [get_ports FMC1_HPC_LA23_P]
+set_property PACKAGE_PIN P30 [get_ports FMC1_HPC_LA23_P]
+
+# extUsb
+set_property IOSTANDARD LVCMOS18 [get_ports FMC1_HPC_LA28_N]
+set_property PACKAGE_PIN L30 [get_ports FMC1_HPC_LA28_N]
+set_property IOSTANDARD LVCMOS18 [get_ports FMC1_HPC_LA28_P]
+set_property PACKAGE_PIN L29 [get_ports FMC1_HPC_LA28_P]
+set_property IOSTANDARD LVCMOS18 [get_ports FMC1_HPC_LA29_N]
+set_property PACKAGE_PIN T30 [get_ports FMC1_HPC_LA29_N]
+set_property IOSTANDARD LVCMOS18 [get_ports FMC1_HPC_LA29_P]
+set_property PACKAGE_PIN T29 [get_ports FMC1_HPC_LA29_P]
+# }}} FMC1
+
 #################
 # clock groups
 
